Fixed the errors and added MiniORM.App

Alexandr0w opened the pull request:
Hello Kris,
I fixed the errors from DbContext and DbSet. I added MiniORM.App for the framework and added more messages into ErrorMessages.cs.

KrIsKa7a commented:
Hello @Alexandr0w,

Please limit the changes to the fix itself. We do not to overwrite the code in the entire repository. Try to isolate the bug and provide only the fix as a new commit in a new pull request. In order to do so, use the main branch of this repository as a baseline. You can use any diff checking tool for merge of the fix. I hope you find this real-life experience exercise interesting! Thanks for the support!

Alexandr0w commented:
Hello again,

The changes are in:
1. DbContext.cs in method SaveChanges() when called MakeGenericMethod 
Type dbSetType = dbSet.GetType();
Type entityType = dbSetType.GetGenericArguments()[0];

MethodInfo persistMethod = typeof(DbContext)
	.GetMethod("Persist", BindingFlags.NonPublic | BindingFlags.Instance)!
	.MakeGenericMethod(entityType);
	
2. DbContext.cs in method MapRelations where I added console writeline with property names and types

3. DbContext.cs in method MapCollection where I added  
var collectionEntitiesAsList = navCollectionEntities.ToList();
ReflectionHelper.ReplaceBackingField(dbSetEntity, collectionPropInfo.Name, collectionEntitiesAsList);

4. In DbSet.cs I changed this.Entities = entities.ToArray(); --> this.Entities = entities.ToList(); 
And in Add method I created a check for null items.

Alexandr0w commented:
I closed this one, opened a new.
